inspire/modules/workflows/tasks/matching.py

```diff
@@ -248,4 +248,5 @@ def _update_old_object(obj, eng):
 def arxiv_set_category_field(obj, eng):
     """Temporary measure to enable sorting by primary category."""
     record = get_record_from_obj(obj, eng)
-    obj.uri = record.get("report_number.arxiv_category", [""])[0]
+    obj.uri = record['arxiv_eprints'][0]['categories'][0]
```

Comment:
I think this could still be accessed like `record.get("arxiv_eprints.categories". [""])[0]` (as `record` is a `Record` object at this point. If you can try it it would be great. Then you can also remove the commented line below.